applications/dot_marketplace-phase2.md

```diff
@@ -0,0 +1,218 @@
+
+- Here's a link to the [approved grant proposal for the first phase.](https://github.com/w3f/Grants-Program/blob/master/applications/dot_marketplace.md)
+- We are in touch with Marcin and Raul from the Web 3 Grants and Treasuries team, respectively.
+
+## **Development Roadmap** 🔩
+
+****Overview****
+
+* **Total Estimated Duration:** 3 Weeks
+* **Full-Time Equivalent (FTE):**  3.36
+* **Total Costs:** 24,305 USD
+
+
+### **Milestone 1**
+
+* **Estimated duration:** 1 week
+* **FTE:**  2
+* **PTE:**  2
+* **Costs:**  8,325 USD  
+
+The main deiverable for this milestone will be to migrate the existing application to substrate frame v2 and include the chat feature as a pallet for communication between a customer and a service provider to have a one-on-one conversation over the deliverables and timelines. The entire milestone covers the Rust/Substrate code implementation.
+
+| Sr no. | Deliverable | Description |
+| --- | --- | --- |
+| 0a | License | Apache 2.0 |
+| 0b | Documentation | We will provide both inline documentation of the code and a tutorial that explains how a user can use DOT Marketplace and understand the flow of tasking pallet. |
+| 0c | Testing Guide | Functions will be covered by unit tests, the documentation will describe how to run these tests. We will also provide scripts to help deploy, run and test the build. |
+| 0d | Docker Image | Docker image of the build |
+| 1 | Migration from frame v1 to frame v2 | The [existing application backend](https://github.com/WowLabz/tasking_backend/blob/Phase1_Milestone3/pallets/pallet-tasking/src/lib.rs) will be migrated to frame v2 to account for the new features provided by the framework |
+| 2 | Chat Pallet | Chat functionality is to be exposed and consumed between the customer and the service provider to ease communication and this will be integrated with the tasking pallet's frame v2 version |
+|  |  |  |
+
+### **Milestone 2**
+
+* **Estimated duration:** 2 weeks
+* **FTE:**  2
+* **PTE:**  2
+* **Costs:**  15,900 USD  
+
+
+In continuation to previous work, this milestone involves creation of a on chain decentralized court to handle dispute resolution. Being a juror is one of the user incentives that can be achieved thanks to the rating module mentioned in the first phase of dot marketplace. The entire milestone covers the Rust/Substrate code implementation.
+
+| Sr no. | Deliverable | Description |
+| --- | --- | --- |
+| 0a | License | Apache 2.0 |
+| 0b | Documentation | We will provide both inline documentation of the code and a tutorial that explains how a user can use DOT Marketplace and understand the flow of tasking pallet. |
+| 0c | Testing Guide | Functions will be covered by unit tests, the documentation will describe how to run these tests. We will also provide scripts to help deploy, run and test the build. |
+| 0d | Docker Image | Docker image of the build |
```

Comment:
Also here: What will the docker image contain?

Reply:
> Also here: What will the docker image contain?

The docker image will contain the entire tasking pallet Frame V2 version for anybody to just deploy it on their terminal without cloning the repo explicitly, it will be explained along with the commands for testing and running the image.

Reply:
> to just deploy it on their terminal without cloning the repo explicitly

So by that you mean you'll upload it to a docker registry like docker hub?!

Feel free to add this info also to the grant application. In general, the application should contain all the details relevant to the application.

Reply:
@takahser Updated the application document. Please check and let me know.